MWI: Add Bound Keypair Joining admin and reference guide

timothyb89 opened the pull request:
This adds an admin and reference guide for bound keypair joining, intended to explain the core architecture of the join method, provide a few minimal configuration examples, and cover various long term maintenance tasks expected to be needed for bots using the `bound_keypair` join method, like rotating keypairs, recovering broken bots, and making other config changes.

timothyb89 committed:
MWI: Add Bound Keypair Joining admin and reference guide

This adds an admin and reference guide for bound keypair joining,
intended to explain the core architecture of the join method, provide
a few minimal configuration examples, and cover various long term
maintenance tasks expected to be needed for bots using the
`bound_keypair` join method, like rotating keypairs, recovering
broken bots, and making other config changes.

ptgott commented on docs/pages/reference/machine-id/bound-keypair.mdx:
```diff
@@ -0,0 +1,587 @@
+---
+
+Bound Keypair is a join method designed to provide the best features of
+[delegated join methods][s-vs-d] join methods - like AWS, GCP, or Azure - but in
```
```suggestion
[delegated join methods][s-vs-d] - like the AWS, GCP, and Azure join methods - but in
```

ptgott commented on docs/pages/reference/machine-id/bound-keypair.mdx:
```diff
@@ -0,0 +1,587 @@
+Bound Keypair Joining is available in v18.1.0 and is intended to replace `token`
+joining as the default recommended join method in Teleport v19.0.0.
+</Admonition>
+
```
I would describe the scope of this guide here. This is especially important since the guide includes multiple purposes in a single document—it might be difficult for readers to determine whether this guide is right for them.

ptgott reviewed (commented):
I would start by adding a description in the introductory section of what this guide is about. This will help users determine whether this guide is the one they should carry on reading. Otherwise, since there are so many topics in this guide, it'll be difficult for a user to determine this.

I also think adding a scope paragraph will help us determine if we should keep this as one guide. Part of me feels like we should take the following steps to split up the guide while keeping it in the same location of the docs:
1. Create an overview page at `docs/pages/reference/machine-id/bound-keypair/bound-keypair.mdx` that includes the use cases and limitations sections of this guide, plus a list of guides in the Bound Keypair section.
1. Move the existing how-to guide to `docs/pages/reference/machine-id/bound-keypair/getting-started.mdx`
1. Move the Concepts section into `docs/pages/reference/machine-id/bound-keypair/concepts.mdx`

I'm not sure about the admin reference—would it make sense to move this to the final section of the how-to guide? That's where we typically include practical steps that don't fit into the main flow of the setup we want users to follow.

timothyb89 committed:
Split admin guide and concepts into separate pages

Also, move the Getting Started guide to the bound keypair reference
section.

timothyb89 reviewed (commented):
I've split the guide into a dedicated Concepts page, moved the Getting Started guide into the reference section, and also split out the admin guide into an individual page.

> I'm not sure about the admin reference—would it make sense to move this to the final section of the how-to guide? That's where we typically include practical steps that don't fit into the main flow of the setup we want users to follow.

I've ended up giving it a dedicated page - my feeling was that it felt a bit odd to have long-term reference material as part of the getting started guide? But if that's the established pattern and what users will expect I'm happy to move things around.

ptgott reviewed (approved):
This seems like a good first iteration to me. Approved, though I think it'd be good to find a solution to the comment here: https://github.com/gravitational/teleport/pull/56824/files#r2224128538

timothyb89 committed:
Remove token examples and add preregistered key flow to guide

This removes the token examples from the bound keypair overview page
and instead moves the key unique information (preregistered keys)
into an alternative flow section in the bound keypair guide.

timothyb89 committed:
Add missing redirect

timothyb89 committed:
Fix broken header links after reorganization
